test/backendSpec.js

```diff
@@ -0,0 +1,249 @@
+'use strict'
+
+const chai = require('chai')
+chai.use(require('chai-as-promised'))
+const assert = chai.assert
+
+const getObjBackend = require('../src/util/backend')
+const ObjStore = require('./helpers/objStore')
+
+describe('ObjStore and ObjBackend', function () {
+  beforeEach(function () {
+    this.store = new ObjStore()
+    this.backend = getObjBackend(this.store)
+  })
+
+  describe('maxValueTracker', function () {
+    beforeEach(function () {
+      this.opts = {
+        key: 'foo'
+      }
+
+      this.tracker = this.backend.getMaxValueTracker(this.opts)
+    })
+
+    it('should return a default value', async function () {
+      assert.deepEqual(
+        await this.tracker.getMax(),
+        { value: '0', data: null })
+    })
+
+    it('should set a higher value', async function () {
+      const last = await this.tracker.setIfMax({ value: '3', data: { foo: 'bar' } })
+
+      assert.deepEqual(last, { value: '0', data: null })
+      assert.deepEqual(
+        await this.tracker.getMax(),
+        { value: '3', data: { foo: 'bar' } })
+
+      // key is concatenated to mvt_maximum
+      assert.deepEqual(
+        JSON.parse(await this.store.get('foo:mvt:maximum')),
+        { value: '3', data: { foo: 'bar' } })
+    })
+
+    it('should not set for a lower value', async function () {
+      const last = await this.tracker.setIfMax({ value: '-1', data: { foo: 'bar' } })
+
+      assert.deepEqual(last, { value: '-1', data: { foo: 'bar' } })
+      assert.deepEqual(
+        await this.tracker.getMax(),
+        { value: '0', data: null })
+    })
+
+    it('should load a value from the store', async function () {
+      await this.tracker.setIfMax({ value: '1', data: 'foo' })
+
+      // uses same key as this.tracker
+      const newTracker = this.backend.getMaxValueTracker(this.opts)
+
+      assert.deepEqual(
+        await newTracker.getMax(),
+        { value: '1', data: 'foo' })
+    })
+
+    it('should not overlaod a separate key', async function () {
+      await this.tracker.setIfMax({ value: '1', data: 'foo' })
+      const newTracker = this.backend.getMaxValueTracker({
+        key: 'bar'
+      })
+
+      assert.deepEqual(
+        await this.tracker.getMax(),
+        { value: '1', data: 'foo' })
+
+      assert.deepEqual(
+        await newTracker.getMax(),
+        { value: '0', data: null })
+    })
+
+    it('should not allow an invalid key', async function () {
+      assert.throws(() => this.backend.getMaxValueTracker({ key: 'test:' }),
+        /invalid key: test:/)
+    })
+
+    it('handles an invalid value', async function () {
+      await assert.isRejected(this.tracker.setIfMax({ foo: 'bar' }),
+        /entry .* must have a value/)
+    })
+  })
+
+  describe('transferLog', function () {
```

Comment:
added tests to make sure that it when ID and content are the same it neither complains nor applies twice, and when ID matches but content doesn't match, it throws an error.